src/cms.js

```diff
@@ -5,12 +5,12 @@ const forge = require('node-forge')
 const util = require('./util')
 
 class CMS {
-  constructor (keystore) {
-    if (!keystore) {
-      throw new Error('keystore is required')
+  constructor (keychain) {
+    if (!keychain) {
+      throw new Error('keychain is required')
     }
 
-    this.keystore = keystore
+    this.keychain = keychain
   }
 
   createAnonymousEncryptedData (name, plain, callback) {
```

Comment:
What is the definition of AnonymousEncryptedData? Why does it need to exist?

Reply:
Will add documentation, real soon.  This creates cryptographically secure data.  That is data that can only be read by the holder of the key.

Reply:
See https://github.com/richardschneider/ipfs-encryption/issues/7

Reply:
How is that different from regular encrypted data?

Reply:
encrypted data is just a blob.  CMS provides structure.  It particular it describes the data encryption algorithm and parameters.  Also it contains a `recipientInfo` that describes the key that can be used to obtain the data encryption key.

Reply:
See https://github.com/richardschneider/ipfs-encryption/issues/9

Reply:
I see, so this function name is more than just describing what it is it actually points to a data structure defined by CMS/PKCS#7. Please make sure to add this as a comment so that others don't get confused. (I'm always concerned on having design info on people's heads)

Reply:
I have written design notes in `richardschneider/ipfs-encryption` please start with https://github.com/richardschneider/ipfs-encryption/issues/3 and https://github.com/richardschneider/ipfs-encryption/issues/3.  Then follow the hyper-links.